rules.json

```diff
@@ -482,6 +500,24 @@
         "code": "function (user, context, callback) {\n  if (!user.app_metadata || !user.app_metadata.authorization ||\n  !Array.isArray(user.app_metadata.authorization.groups)) {\n    return callback(null, user, context);\n  }\n\n  var groups = user.app_metadata.authorization.groups;\n  var GROUPS_WITH_MFA = {\n    // Add groups that need MFA here\n    // Example\n    // 'admins': true,\n  };\n\n  var needsMFA = !!groups.find(function(group) {\n    return GROUPS_WITH_MFA[group];\n  });\n\n  if (needsMFA){\n      context.multifactor = {\n        provider: 'guardian', //required\n\n        ignoreCookie: true, // optional. Force Auth0 MFA everytime this rule runs. Defaults to false. if accepted by users the cookie lasts for 30 days (this cannot be changed)\n      };\n  }\n\n  callback(null, user, context);\n}"
       },
       {
+        "title": "Force multifactor when IP outside an expected range",
+        "categories": [
+          "multifactor",
+          "guardian"
+        ],
+        "description": "<p>This rule is used to trigger multifactor authentication when the requesting IP is from outside the corporate IP range.</p>\n",
+        "code": "function (user, context, callback) {\n  var ipaddr = require('ipaddr.js');\n  var corp_network = '192.168.1.134/26';\n\n  var current_ip = ipaddr.parse(c  ontext.request.ip); \n  if(!current_ip.match(ipaddr.parseCIDR(corp_network))){\n    context.multifactor = {\n        provider: 'guardian',\n        ignoreCookie: true, \n        };  \n  }\n  \n  callback(null, user, context);\n}\n"
+      },
+      {
+        "title": "Force multifactor when country has changed",
+        "categories": [
+          "multifactor",
+          "guardian"
+        ],
+        "description": "<p>This rule is used to trigger multifactor authentication when the user has changed countries since the last request..</p>\n",
+        "code": "function (user, context, callback) {\n\n  user.app_metadata = user.app_metadata || {}; \n  \n  if(user.app_metadata.last_location !== context.request.geo  ip.country_code){\n    context.multifactor = { \n      provider: 'guardian',\n      ignoreCookie: true, \n    };  \n  }\n  \n  //Set the location context for next time\n    user.app_metadata.last_location = context.request.geoip.country_code;\n  auth0.users.updateAppMetadata(user.user_id, user.app_metadata)\n    .then(function(){\n      cal  lback(null, user, context);\n    })  \n    .catch(function(err){\n      callback(err);\n  }); \n}\n"
```

Comment:
Here is a copy paste into vi, followed by :1,$s/\\n/^M/g

function (user, context, callback) {

  user.app_metadata = user.app_metadata || {}; 
  
  if(user.app_metadata.last_location !== context.request.geo  ip.country_code){
    context.multifactor = { 
      provider: 'guardian',
      ignoreCookie: true, 
    };  
  }
  
  //Set the location context for next time
    user.app_metadata.last_location = context.request.geoip.country_code;
  auth0.users.updateAppMetadata(user.user_id, user.app_metadata)
    .then(function(){
      cal  lback(null, user, context);
    })  
    .catch(function(err){
      callback(err);
  }); 
}